### ISSUE TITLE
packages/sw and packages/frontend are not typechecked in CI, and do not typecheck successfully.

### ISSUE THREAD

**Issue by yuchiki** [syuilo/Misskey#10329]
## 💡 Summary

Misskey has three packages, namely backend, frontend and sw.
However, only backend package is typechecked in CI

(see also: https://github.com/misskey-dev/misskey/blob/c05c504c8632591ef18170c2881b0b44b0027ed9/.github/workflows/lint.yml#L63)

As a consequence, these modules' type information is obsoleted now so they don't typecheck. (I checked it with 'pnpm lint' command)

## 🥰 Expected Behavior

These two packages are correctly typed, and typechecked in CI. 

## 🤬 Actual Behavior

N/A 

## 📝 Steps to Reproduce

N/A

## 📌 Environment

N/A

I wrote a PR to fix types in sw, so I will publish it later.

**Comment by syuilo** [comment]
Client-side type checking is intentionally disabled because it is impractical due to many type errors related to the DOM and other factors.

**Comment by saschanaz** [comment]
>Client-side type checking is intentionally disabled because it is impractical due to many type errors related to the DOM and other factors.

I have to disagree as a maintainer of TypeScript DOM library, do you have a specific pain point? IMO errors are mainly coming from Vue use and misskey-js (#10322)

**Comment by acid-chicken** [comment]
~~It's time to ditch vue~~ Type checking is even worth the cost of using ts-ignore frequently.

**Comment by saschanaz** [comment]
> ~~It's time to ditch vue~~

Custom Elements使いたいです

**Comment by yuchiki** [comment]
as a consequence of lacking typecheck in CI, some parts of type information written in the code are simply wrong, and it is not so desirable.

I have a desire to help the development of this project, and at least,  for the first time, i would like to fix the wrong description of types.

**Comment by zyoshoka** [comment]
Related to #10131

**Comment by zyoshoka** [comment]
These packages are now typechecked in CI (https://github.com/misskey-dev/misskey/commit/30d87d5532ca7efe07aa26538d8a87a02be63c73).
